In [1]:
import pandas as pd
import numpy as np
df = pd.read_csv(r"E:\5 Python\property.csv") 
df.head()

Suburb           Address  Rooms Type      Price Method SellerG  \
0  Abbotsford      85 Turner St      2    h  1480000.0      S  Biggin   
1  Abbotsford   25 Bloomburg St      2    h  1035000.0      S  Biggin   
2  Abbotsford      5 Charles St      3    h  1465000.0     SP  Biggin   
3  Abbotsford  40 Federation La      3    h   850000.0     PI  Biggin   
4  Abbotsford       55a Park St      4    h  1600000.0     VB  Nelson   

        Date  Distance  Postcode  ...  Bathroom  Car  Landsize  BuildingArea  \
0  3/12/2016       2.5    3067.0  ...       1.0  1.0     202.0           NaN   
1  4/02/2016       2.5    3067.0  ...       1.0  0.0     156.0          79.0   
2  4/03/2017       2.5    3067.0  ...       2.0  0.0     134.0         150.0   
3  4/03/2017       2.5    3067.0  ...       2.0  1.0      94.0           NaN   
4  4/06/2016       2.5    3067.0  ...       1.0  2.0     120.0         142.0   

   YearBuilt  CouncilArea Lattitude  Longtitude             Regionname  \
0        NaN        Yarra  -37.7996    144.9984  Northern Metropolitan   
1     1900.0        Yarra  -37.8079    144.9934  Northern Metropolitan   
2     1900.0        Yarra  -37.8093    144.9944  Northern Metropolitan   
3        NaN        Yarra  -37.7969    144.9969  Northern Metropolitan   
4     2014.0        Yarra  -37.8072    144.9941  Northern Metropolitan   

  Propertycount  
0        4019.0  
1        4019.0  
2        4019.0  
3        4019.0  
4        4019.0  

[5 rows x 21 columns]

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13580 entries, 0 to 13579
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         13580 non-null  object 
 1   Address        13580 non-null  object 
 2   Rooms          13580 non-null  int64  
 3   Type           13580 non-null  object 
 4   Price          13580 non-null  float64
 5   Method         13580 non-null  object 
 6   SellerG        13580 non-null  object 
 7   Date           13580 non-null  object 
 8   Distance       13580 non-null  float64
 9   Postcode       13580 non-null  float64
 10  Bedroom2       13580 non-null  float64
 11  Bathroom       13580 non-null  float64
 12  Car            13518 non-null  float64
 13  Landsize       13580 non-null  float64
 14  BuildingArea   7130 non-null   float64
 15  YearBuilt      8205 non-null   float64
 16  CouncilArea    12211 non-null  object 
 17  Lattitude      13580 non-null  float64
 18  Longti

## Q4 : In the suburb of Abbotsford, what are the chances of finding a property with 3 rooms? Round your answer to 3 decimal places. 

In [3]:
abbotsford_data = df[df['Suburb'] == 'Abbotsford']# Filter data for Abbotsford

total_properties = len(abbotsford_data)# Count total properties in Abbotsford

properties_with_3_rooms = len(abbotsford_data[abbotsford_data['Rooms'] == 3]) # Count properties with 3 rooms

probability = properties_with_3_rooms / total_properties # Calculate the probability

#probability_rounded = round(probability, 3)# Round off the result to 3 decimal places
print(f"Total properties in Abbotsford: {total_properties}")
print(f"Properties with 3 Rooms: {properties_with_3_rooms}")
print(f"Probability of finding a property with 3 rooms: {probability:.3f}") # showing and Rounding answer to 3 decimal places

Total properties in Abbotsford: 56
Properties with 3 Rooms: 20
Probability of finding a property with 3 rooms: 0.357


**Conclusion:**
- Among all properties in Abbotsford (56 in total), a significant portion (20 out of 56) consists of 3-room properties. This makes 3-room properties relatively common.
- If you randomly select a property in Abbotsford, the chance of it having 3 rooms is approximately 35.7%.


## Q5 : In the suburb of Abbotsford, what are the chances of finding a property with 2 bathrooms? Round your answer to 3 decimal places.

In [4]:
abbotsford_data = df[df['Suburb'] == 'Abbotsford'] # Filter data for Abbotsford

total_properties = len(abbotsford_data) # Count total properties in Abbotsford

properties_with_2_bathrooms = len(abbotsford_data[abbotsford_data['Bathroom'] == 2]) # Count properties with 2 bathrooms

probability = properties_with_2_bathrooms / total_properties # Calculate the probability
print(f"Total properties in Abbotsford: {total_properties}")
print(f"Properties with 2 bathrooms: {properties_with_2_bathrooms}")
print(f"Probability of finding a property with 2 bathrooms: {probability:.3f}") # showing and Rounding answer to 3 decimal places

Total properties in Abbotsford: 56
Properties with 2 bathrooms: 19
Probability of finding a property with 2 bathrooms: 0.339


**Conclusion:**
- Among all properties (56 in total), a considerable portion (19 out of 56) has 2 bathrooms, making them relatively common but slightly less frequent than 3-room properties.
- If you randomly select a property in Abbotsford, the chance of it having 2 bathrooms is 33.9%.
